### ISSUE TITLE
[Instability] Gradient Norm and Loss "Vibration" (Oscillation) during Qwen3-VL-8B Finetuning

### ISSUE THREAD

**Issue by TedLiangHK** [NVIDIA-NeMo/Megatron-Bridge#1950]
Description
I am encountering a severe instability pattern ("vibration" or sawtooth wave) in the LM Loss and Gradient Norm when finetuning Qwen3-VL-8B-Instruct. The Gradient Norm oscillates violently between standard values (~10–20) and massive spikes (>500), constantly hitting the gradient clipper.

System & Environment
Model: Qwen3-VL-8B-Instruct

Megatron-bridge code revision:
commit 5c7ebe7d8c31e75ddf882e1e6ced69d10e250786 and applied patch in https://github.com/NVIDIA-NeMo/Megatron-Bridge/issues/1776

Hardware: NVIDIA GPUs (High utilization ~750 TFLOPs observed)

Precision: BF16 (loss_scale: 1.0)

Key Arguments from Log:

YAML

global_batch_size: 64        # Likely too small for multimodal variance
freeze_vision_model: false   # Unfrozen vision tower adds gradient noise
clip_grad: 1.0               # Aggressively clipping the spikes
dataloader_type: single      # Sequential processing?
Logs & Evidence
The training logs show the Gradient Norm spiking by 10x–50x within a few iterations, creating the vibration pattern:

Plaintext

[2026-01-14 10:17:53] iteration 2 | global batch size: 64 | grad norm: 864.104  <-- Massive Spike
[2026-01-14 10:18:04] iteration 4 | global batch size: 64 | grad norm: 198.975
...
[2026-01-14 10:44:08] iteration 326 | global batch size: 64 | grad norm: 475.674 <-- Spike
[2026-01-14 10:44:13] iteration 327 | global batch size: 64 | grad norm: 101.331
...
[2026-01-14 16:56:06] iteration 4998 | global batch size: 64 | grad norm: 14.22

Full log and "lm-loss" and "grad-norm" diagram attached

[training_10_16_Jan_14.log](https://github.com/user-attachments/files/24630530/training_10_16_Jan_14.log)
<img width="3560" height="3537" alt="Image" src="https://github.com/user-attachments/assets/ad123381-80a3-4837-8018-d83e5283894a" />

**Comment by TedLiangHK** [comment]
The command used is as below:
 /opt/venv/bin/python -m torch.distributed.run --nproc_per_node=8 examples/recipes/qwen_vl/finetune_qwen_vl.py \
    --recipe qwen3_vl_8b_finetune_config \
    --hf_path /workspace/models/Qwen3-VL-8B-Instruct \
    --pretrained-checkpoint /workspace/models/Qwen3-VL-8B-Instruct-Nemo \
    --dataset-type preloaded \
    --data-path /workspace/dataset/32B_cloudflare_datasets/stage_II/sharegpt_format/ScaleCUA.json  \
    model.tensor_model_parallel_size=2 \
    model.pipeline_model_parallel_size=1 \
    train.micro_batch_size=1 train.global_batch_size=64 \
    model.seq_length=8192 \
    model.freeze_language_model=False model.freeze_vision_model=False \
    train.train_iters=5000 train.eval_interval=2000 checkpoint.save_interval=1000 \
    optimizer.lr=1.5e-5 optimizer.min_lr=1e-7 \
    scheduler.lr_warmup_iters=500 scheduler.lr_decay_iters=5000 \
    logger.log_interval=1

**Comment by yaoyu-33** [comment]
hi it's a bit hard to see where the issue comes from since we dont have the same dataset.   I wonder if you already play with the recipe a bit, like reduce lr and maybe increase global batch. 
does freeze vision helps reduce the stability?

**Comment by TedLiangHK** [comment]
I have tried reducing lr and increasing micro-batch size and global batch size, and it doesn't help with the oscillation.
My dataset is composed of samples like below:
[
  {
    "from": "human",
    "value": "You are a Desktop Agent that can take screenshots and locate GUI elements based on user instructions.\n\nYou will be provided with a screenshot in the beginning, and in each round you will be given an instruction. You need to locate the target by giving a point coordinate (x, y), where x and y are integers normalized between 0 and 1000.\n\nThe screenshot is:\n<image>\n\nThe instruction is:\nClick on 'quick_start.py' in the project file tree to open it in the editor.\n"
  },
  {
    "from": "gpt",
    "value": "(56, 385)"
  },
  {
    "from": "human",
    "value": "Select the quick_start.py file from the project explorer to view its contents."
  },
  {
    "from": "gpt",
    "value": "(56, 385)"
  },
  {
    "from": "human",
    "value": "Open the quick_start.py Python script by clicking on its name in the file navigation panel."
  },
  {
    "from": "gpt",
    "value": "(56, 385)"
  },
  {
    "from": "human",
    "value": "The 'quick_start.py' Python script file entry in the project file explorer sidebar of the IDE."
  },
  {
    "from": "gpt",
    "value": "(56, 385)"
  }
]

**Comment by weijiac0619** [comment]
Thanks for the details. A few follow-ups to help narrow this down:
 1. Does freezing the vision tower (model.freeze_vision_model=True) reduce the oscillation?
 2. Can you reproduce on the current main?
